### DISCUSSION TITLE
How do you guys structure your application when using zod?

### DISCUSSION THREAD

**Discussion by lewisd1996** [colinhacks/zod#1663] [General]
Just tried out zod and I love it. I would like to implement it across my React application. 

An example of component folder in my app looks like:

<img width="317" alt="Screenshot 2022-12-10 at 11 16 23" src="https://user-images.githubusercontent.com/25225528/206852289-cd8941b3-4cfc-4e2e-8c06-265158b001f3.png">

So each component has a `components`, `helpers`, `types` etc folders.

Do you guys do something simialr and create a `schema` or a `zod` folder? And just export the entire schema from there, allowing you to use it elsewhere? Or do something different, like store them all at a global level?  

Looking to see how others implement `zod` into their application in a clean way, as most tutorials just declare their schemas in the file.

**Comment by JacobWeisenburger** [comment]
My approach to folder structures is that it's based entirely on the goals/needs of the project/team. So I don't have a structure that I use everywhere. I'm guessing this doesn't help that much, but I wanted to answer your question so you didn't feel ignored. Sorry that I couldn't be of more help. If you have a more specific question, I could probably give a better answer.

**Comment by pauldvu** [comment]
Things are always changing but right now my implementation in is to put everything in a types folder then  define my exported schemas and exported types. After trying out a organized folder structure I found myself wishing  I had put everything together into one file. 

I currently have a mono repo so this helps a lot as I just import my types to both the back end and front end from the single file. 

```
/common/types
/backend
/frontend 
```

**Reply by AaaSsHhUu** [reply]
i also tried this, but there is a little issue when i am importing zod types from common to my backend the tsconfig file gives me an error that common folder is not inside src as the rootDir is set to  './src' of backend. 
How can i resolve this???

**Comment by Mohammed-Hammood** [comment]
Try FSD [(Feature-sliced design)  ](https://feature-sliced.design/docs/get-started/overview)

**Comment by joywin2003** [comment]
My practice is to place schemas inside /lib/schema/index.ts, and I believe this is an appropriate approach. While it is a matter of personal preference, logically speaking, it makes sense to store them in the lib or utils directory, as these directories are meant for libraries or utilities used throughout the project. I hope this clarifies my reasoning.

**Reply by JoshuaDeveloper25** [reply]
That's a good point.
